### PR TITLE
Testing library component

### PR DESCRIPTION
**Pull request by CesarValiente**
This branch/PR basically takes the `test-utils` module, that we have been using to help implementing our UI tests, and make it available as a standalone library.

I've deleted some functionality that was not used (e.g. Mock helper), and refactored and added new one like Surface Duo 2 dimensions, FoldingFeature utilities, and made specific functionality more generic so that other modules could use it too (+ can be used targeting other foldable devices).

I've changed the package name to be aligned with our [Jetpack Compose Testing library](https://github.com/microsoft/surface-duo-compose-sdk/tree/main/ComposeTesting), and added publication bits needed to build a new artifact and make it available on the artifact feeds we use.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,6 +86,12 @@ When importing the code into Android Studio, use the **surface-duo-sdk** folder
     implementation "com.microsoft.device.dualscreen:navigation-ui-ktx:$nav_version"
     ```
 
+   **Testing**
+
+    ```gradle
+    implementation "com.microsoft.device.dualscreen.testing:testng-kotlin:1.0.0-alpha1"
```
Comment (bimiron) [inline README.md:92]: I think there is a typo here, missing an "i" in  `:testng-kotlin:`

File: utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/ViewMatcher.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+ * @param firstDisplayWith : width of the left/start/first display area
+ * @param totalDisplayWith : width of the right/end/second/total display area
+ * @param foldingFeatureWidth : width of the FoldingFeature {@see FoldingFeature} if any.
+ * @return
+ */
+fun isViewOnScreen(
+    position: DisplayPosition,
+    orientation: Int,
+    firstDisplayWith: Int,
+    totalDisplayWith: Int,
```
Comment (bimiron) [inline utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/ViewMatcher.kt:29]: I think there are 2 typos here - `firstDisplayWith ` and `totalDisplayWith`. Should be Width, right?

File: utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/ViewMatcher.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+ * @param totalDisplayWith : the width of the total display area.
+ * @param foldingFeatureWidth : the width of the folding feature (if any).
+ * @return
+ */
+fun areCoordinatesOnTargetScreen(
+    targetScreenPosition: DisplayPosition,
+    xStart: Int,
+    xEnd: Int,
+    firstDisplayWith: Int,
+    totalDisplayWith: Int,
```
Comment (bimiron) [inline utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/ViewMatcher.kt:83]: Same here - I think there are 2 typos here - `firstDisplayWith` and `totalDisplayWith`. Should be Width, right?

File: utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/SurfaceDuoUtils.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    val SINGLE_SCREEN_HINGE_RECT = Rect()
+    val DUAL_SCREEN_HINGE_RECT = Rect(1344, 0, 1410, 1892)
+    const val SINGLE_SCREEN_WIDTH = 1344
+    const val HINGE_WIDTH = 66
+    const val SCREEN_COUNT = 2
+    const val DUAL_SCREEN_WIDTH = SINGLE_SCREEN_WIDTH * SCREEN_COUNT + HINGE_WIDTH
+}
+
+/**
+ * Utility function that switches the  application from single screen mode to dual screen on Surface Duo devices.
```
Comment (bimiron) [inline utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/SurfaceDuoUtils.kt:48]: The values for span and un-span using `device.swipe` are only for SurfaceDuo1, should we mention that somewhere?

**Review by bimiron — APPROVED** [review]
Great work, Cesar! Just added some small suggestions:)

File: README.md (hunk 1 above)
Comment (CesarValiente) [reply README.md:92]: 😱😱😱 good catch! thanks!

File: utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/ViewMatcher.kt (hunk 2 above)
Comment (CesarValiente) [reply utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/ViewMatcher.kt:29]: 😨😱😱😱 totally!! thanks Bianca!

File: utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/ViewMatcher.kt (hunk 3 above)
Comment (CesarValiente) [reply utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/ViewMatcher.kt:83]: 🥵😢 Totally! Thanks!

File: utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/SurfaceDuoUtils.kt (hunk 4 above)
Comment (CesarValiente) [reply utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/SurfaceDuoUtils.kt:48]: Fantastic catch! I will mention that! Thanks!

**Review by soalb-m — APPROVED** [review]

File: utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/SurfaceDuoUtils.kt (hunk 4 above)
Comment (bimiron) [reply utils/test-utils/src/main/java/com/microsoft/device/dualscreen/testing/SurfaceDuoUtils.kt]: Looks great!

**Pull request merged**